tests/sampling/test_forward.py

```diff
@@ -427,6 +428,36 @@ def test_mutable_coords_volatile(self):
             "offsets",
         }
 
+    def test_length_coords_volatile(self):
+        with pm.Model() as model:
+            model.add_coord("trial", length=3)
+            x = pm.Normal("x", dims="trial")
+            y = pm.Deterministic("y", x.mean())
+
+        # Same coord length -- `x` is not volatile
+        trace_same_len = az_from_dict(
+            posterior={"x": [[[np.pi] * 3]]},
+            coords={"trial": range(3)},
+            dims={"x": ["trial"]},
+        )
+        with model:
+            pp_same_len = pm.sample_posterior_predictive(
+                trace_same_len, var_names=["y"]
+            ).posterior_predictive
+        assert pp_same_len["y"] == np.pi
+
+        # Coord length changed -- `x` is volatile
```

Comment:
I prefer to change the dim length in the pymc model and pass exactly the same idata to sample_posterior_predictive as was done in the first call.

Reply:
The way to update an existing dim is with `Model.set_dim(), which requires a new_length`

Reply:
Sorry, just got to this now. Implemented in https://github.com/pymc-devs/pymc/pull/7381/commits/49bbf8c68f44bf25454598af41096dabbcd001ad -- hope I understood you correctly

Reply:
Yup you got it!